cpp/calculate-demand.cpp

```diff
@@ -72,68 +73,88 @@ long readUnsatisfiedDemand(std::string targetFilename) {
+
+  // generate target dataset based on demoDataset
   GDALDataset* demoDataset = openRaster(demoFilename);
   GDALRasterBand* demoBand = demoDataset->GetRasterBand(1);
   CPLAssert(demoBand->GetRasterDataType() == GDT_Float32);
-
   int xBlockSize, yBlockSize;
   demoBand->GetBlockSize(&xBlockSize, &yBlockSize);
-
   GDALDataset* targetDataset = copyRaster(demoDataset, targetFilename, xBlockSize, yBlockSize);
   GDALRasterBand* targetBand = targetDataset->GetRasterBand(1);
   closeRaster(demoDataset);
 
-  targetBand->GetBlockSize(&xBlockSize, &yBlockSize);
-  int xSize = targetDataset->GetRasterXSize();
-  int ySize = targetDataset->GetRasterYSize();
-  int nXBlocks = (xSize + xBlockSize - 1)/xBlockSize;
-  int nYBlocks = (ySize + yBlockSize - 1)/yBlockSize;
+
+  float* buffer = (float*) CPLMalloc(sizeof(float)*xBlockSize*yBlockSize);
+  BYTE* facilityBuffer = (BYTE*) CPLMalloc(sizeof(BYTE)*xBlockSize*yBlockSize);
+
+  float nodata = targetBand->GetNoDataValue();
+
+  int targetXSize = targetDataset->GetRasterXSize();
+  int targetYSize = targetDataset->GetRasterYSize();
+  int targetNXBlocks = (targetXSize + xBlockSize - 1)/xBlockSize;
+  int targetNYBlocks = (targetYSize + yBlockSize - 1)/yBlockSize;
   int nXValid, nYValid, xOffset, yOffset;
 
 #ifdef DEBUG
   std::cerr << "Target raster properties:" << std::endl;
-  std::cerr << " xSize " << xSize << std::endl;
-  std::cerr << " ySize " << ySize << std::endl;
+  std::cerr << " xSize " << targetXSize << std::endl;
+  std::cerr << " ySize " << targetYSize << std::endl;
   std::cerr << " xBlockSize " << xBlockSize << std::endl;
   std::cerr << " yBlockSize " << yBlockSize << std::endl;
-  std::cerr << " nYBlocks " << nYBlocks << std::endl;
-  std::cerr << " nXBlocks " << nXBlocks << std::endl;
+  std::cerr << " nYBlocks " << targetNYBlocks << std::endl;
+  std::cerr << " nXBlocks " << targetNXBlocks << std::endl;
 #endif
 
-  float* buffer = (float*) CPLMalloc(sizeof(float)*xBlockSize*yBlockSize);
-  BYTE* facilityBuffer = (BYTE*) CPLMalloc(sizeof(BYTE)*xBlockSize*yBlockSize);
-
-  float nodata = targetBand->GetNoDataValue();
+  for (size_t iFacility = 0; iFacility < facilities.size(); iFacility++) {
 
-  for (int iFacility = 0; iFacility < (int)facilities.size(); iFacility++) {
 #ifdef DEBUG
     std::cerr << "Processing facility " << facilities[iFacility] << std::endl;
 #endif
+
+    // open isochrone dataset
     GDALDataset* facilityDataset = openRaster(facilities[iFacility]);
     GDALRasterBand* facilityBand = facilityDataset->GetRasterBand(1);
     BYTE facilityNodata = facilityBand->GetNoDataValue();
-
     CPLAssert(facilityBand->GetRasterDataType() == GDT_Byte);
-    int xBlockSizeFacility, yBlockSizeFacility;
-    facilityBand->GetBlockSize(&xBlockSizeFacility, &yBlockSizeFacility);
-    CPLAssert(xBlockSizeFacility == xBlockSize);
-    CPLAssert(yBlockSizeFacility == yBlockSize);
```

Comment:
Let's keep the assertions on blocksize, as they should still match between all rasters